it/controllers/RegimeAPIControllerISpec.scala

```diff
@@ -180,108 +267,189 @@ class RegimeAPIControllerISpec
-//               |}
-//               |""".stripMargin
-//
-//           mockStubResponseForAuthorisedUser
-//           mockStubResponseForGetPenaltyDetails(Status.NOT_FOUND, regime, idType, id, body = Some(notFoundResponseBody))
-//
-//           val result = await(buildClientForRequestToApp(uri = s"/${regime.value}/summary/${idType.value}/${id.value}").get())
-//           result.status shouldBe NO_CONTENT
-//         }
-//
-//         "the get penalty details call returns 200 with an empty body" in {
-//           val emptyResponse: String = "{}"
-//
-//           mockStubResponseForAuthorisedUser
-//           mockStubResponseForGetPenaltyDetails(Status.OK, regime, idType, id, body = Some(emptyResponse))
-//
-//           val result = await(buildClientForRequestToApp(uri = s"/${regime.value}/summary/${idType.value}/${id.value}").get())
-//           result.status shouldBe NO_CONTENT
-//         }
-//       }
-//     }
+      Table(
+      ("Regime", "IdType", "Id"),
+      (Regime("VATC"), IdType("VRN"), Id("123456789")),
+      (Regime("ITSA"), IdType("NINO"), Id("AB123456C"))
+    ).forEvery { (regime, idType, id) =>
+
+      val enrolmentKey = AgnosticEnrolmentKey(regime, idType, id)
+
+    s"getSummaryData for $regime" when {
+      val penaltyUpstreamServices = Seq("HIP", "ETMP")
+      
+      penaltyUpstreamServices.foreach { upstreamService =>
+        def mockHIPSummary(responseStatus: Int): StubMapping = mockResponseForHIPPenaltyDetails(
+          responseStatus,
+          regime,
+          idType,
+          id,
+          body = Some(getHIPPenaltyDetailsJson.toString()))
+
+        def mockETMPSummary(responseStatus: Int, body: Option[String] = None): StubMapping = mockResponseForGetPenaltyDetails(
+          responseStatus,
+          regime,
+          idType,
+          id.value,
+          body)
+
+
+
+        val expectedSummaryResponse = if (upstreamService == "HIP") {
+          Json.parse("""
+            |{
+            |  "noOfPoints": 2,
+            |  "noOfEstimatedPenalties": 1,
+            |  "noOfCrystalisedPenalties": 1,
+            |  "estimatedPenaltyAmount": 246.9,
+            |  "crystalisedPenaltyAmountDue": 144,
+            |  "hasAnyPenaltyData": true
+            |}
+            |""".stripMargin)
+        } else {
+          Json.parse("""
+            |{
+            |  "noOfPoints": 2,
+            |  "noOfEstimatedPenalties": 2,
+            |  "noOfCrystalisedPenalties": 2,
+            |  "estimatedPenaltyAmount": 246.9,
+            |  "crystalisedPenaltyAmountDue": 288,
+            |  "hasAnyPenaltyData": true
+            |}
+            |""".stripMargin)
+        }
+
+        val uriToSummaryController = s"/${regime.value}/summary/${idType.value}/${id.value}"
+
+        def setSummaryFeatureSwitch: Unit =
+          if (upstreamService == "HIP") {
+            setEnabledFeatureSwitches(CallAPI1812HIP)
+          } else {
+            disableFeatureSwitch(CallAPI1812HIP)
```

Comment:
setEnabledFeatureSwitches disables all but the switch given, so there shouldn't be any difference if you remove             disableFeatureSwitch(CallAPI1812HIP)